quickstartKotlin/src/main/java/com/twilio/video/quickstart/kotlin/VideoActivity.kt

```diff
@@ -0,0 +1,669 @@
+     */
+    private fun showConnectDialog() {
+        val roomEditText = EditText(this)
+        alertDialog = createConnectDialog(roomEditText,
+                connectClickListener(roomEditText), cancelConnectDialogClickListener(), this)
+        alertDialog!!.show()
+    }
+
+    /*
+     * Called when participant joins the room
+     */
+    private fun addParticipant(participant: Participant) {
+        /*
+         * This app only displays video for one additional participant per Room
+         */
+        if (thumbnailVideoView.visibility == View.VISIBLE) {
+            Snackbar.make(connectActionFab,
+                    "Multiple participants are not currently support in this UI",
+                    Snackbar.LENGTH_LONG)
+                    .setAction("Action", null).show()
+            return
+        }
+        participantIdentity = participant.identity
+        videoStatusTextView.text = "Participant $participantIdentity joined"
+
+        /*
+         * Add participant renderer
+         */
+        if (participant.videoTracks.isNotEmpty()) {
+            addParticipantVideo(participant.videoTracks.first())
+        }
+
+        /*
+         * Start listening for participant events
+         */
+        participant.setListener(participantListener)
+    }
+
+    /*
+     * Set primary view as renderer for participant video track
+     */
+    private fun addParticipantVideo(videoTrack: VideoTrack) {
+        moveLocalVideoToThumbnailView()
+        primaryVideoView.mirror = false
+        videoTrack.addRenderer(primaryVideoView)
+    }
+
+    private fun moveLocalVideoToThumbnailView() {
+        if (thumbnailVideoView.visibility == View.GONE) {
+            thumbnailVideoView.visibility = View.VISIBLE
+            localVideoTrack?.removeRenderer(primaryVideoView)
+            localVideoTrack?.addRenderer(thumbnailVideoView)
+            localVideoView = thumbnailVideoView
+            thumbnailVideoView.mirror = cameraCapturerCompat.cameraSource ==
+                    CameraCapturer.CameraSource.FRONT_CAMERA
+        }
+    }
+
+    /*
+     * Called when participant leaves the room
+     */
+    private fun removeParticipant(participant: Participant) {
+        videoStatusTextView.text = "Participant " + participant.identity + " left."
+        if (participant.identity != participantIdentity) {
+            return
+        }
+
+        /*
+         * Remove participant renderer
+         */
+        if (participant.videoTracks.size > 0) {
+            removeParticipantVideo(participant.videoTracks[0])
+        }
+        moveLocalVideoToPrimaryView()
+    }
+
+    private fun removeParticipantVideo(videoTrack: VideoTrack) {
+        videoTrack.removeRenderer(primaryVideoView)
+    }
+
+    private fun moveLocalVideoToPrimaryView() {
+        if (thumbnailVideoView.visibility == View.VISIBLE) {
+            thumbnailVideoView.visibility = View.GONE
+            localVideoTrack?.removeRenderer(thumbnailVideoView)
```

Comment:
Multiple method invocations can be called using function [with](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/with.html).